Feature/tms 5 home tasks

KonstantsinYurkevich opened the pull request.

KonstantsinYurkevich committed:
add Search tests by Author name, Surname, Name and surname
Fixes :
1. Deleted useless code strings in search test ( no need to log in in order to test field SEARCH)
2. Replaced method sleep in tests to  driver.manage().timeouts().implicitlyWait()
3. Replaced code to 300 strings in one file (Search tests)

KonstantsinYurkevich committed:
add Search tests by Author name, Surname, Name and surname
Fixes :
1. Deleted useless code strings in search test
2. Replaced code to 300 strings in one file (Search tests)
3. Deleted TestSearchAuthorNameAndSurnameSeparate

KonstantsinYurkevich committed:
Add test for TestSearchNameOfTheBook

KonstantsinYurkevich committed:
Add tests:
1. For Shopping cart (add,prise)
2. Greeting massage
Fixed tests :
1. Login
2. TestSearchAuthorNameAndSurname

KonstantsinYurkevich committed:
Added to the tests driver.manage().timeouts().implicitlyWait(5000, TimeUnit.SECONDS);

KonstantsinYurkevich committed:
Added :
1. HomeTask test
2. Test of discounts
3. Test of discount value
discount value =  (( price * quantity))/100 * discount percent
3. Some Fixes

dzmitryrak commented on src/test/java/chrome/shoppingCart/TestCalculationOfDiscountInValue.java:
```diff
@@ -0,0 +1,113 @@
+        driver.findElement(By.xpath("((//tr)[8]//td)[3]/input")).sendKeys("150");
+        driver.findElement(By.xpath("((//tr)[10]//td)[1]/input")).click();
+        String discountPercent= driver.findElement(By.xpath("((//tr)[9]//td)[6]")).getText();
+        assertEquals(discountPercent,"6000.0","Wrong value of discount");
```
ты добавил 150 книг и discountPercent сравниваешь с неким значением 6000. Не совсем понял что за проверка? Я ожидал что мы сравним реальную общую цену со значением ожидаемым, что-то около 1440

dzmitryrak commented on src/test/java/chrome/shoppingCart/TestCalculationOfDiscountInValue.java:
```diff
@@ -0,0 +1,113 @@
+        driver.findElement(By.xpath("((//tr)[8]//td)[3]/input")).clear();
+        driver.findElement(By.xpath("((//tr)[8]//td)[3]/input")).sendKeys("150");
+        driver.findElement(By.xpath("((//tr)[10]//td)[1]/input")).click();
+        String discountPercent= driver.findElement(By.xpath("((//tr)[9]//td)[6]")).getText();
```
пробелы после запятых забываешь

KonstantsinYurkevich committed:
Fixes:
1. Added spaces after commas
2. Corrected values of discount in TestCalculationOfDiscountInValue:
   a) From 1500 to 15.0
   b) From 6000
3. Correct value in assertEquals(totalPriceReal, "93.00", "Wrong calculation of total price") FROM 10000.00 to 93.0
